packages/phase2cli/src/commands/setup.ts

```diff
@@ -58,15 +68,418 @@ import {
+ * @param zkeyLocalPathAndFileName <string> - the local complete path of the pre-computed zKey selected file.
+ */
+const handleNewZkeyGeneration = async (
+    r1csLocalPathAndFileName: string,
+    potLocalPathAndFileName: string,
+    zkeyLocalPathAndFileName: string
+) => {
+    console.log(
+        `${theme.symbols.info} The computation of your brand new zKey is starting soon.\n${theme.text.bold(
+            `${theme.symbols.warning} Be careful, stopping the process will result in the loss of all progress achieved so far.`
+        )}`
+    )
+
+    // Generate zKey.
+    await zKey.newZKey(r1csLocalPathAndFileName, potLocalPathAndFileName, zkeyLocalPathAndFileName, console)
+
+    console.log(`\n${theme.symbols.success} Generation of genesis zKey completed successfully`)
+}
+
+/**
+ * Manage the creation of a ceremony file storage bucket.
+ * @param firebaseFunctions <Functions> - the Firebase Cloud Functions instance connected to the current application.
+ * @param ceremonyPrefix <string> - the prefix of the ceremony.
+ * @returns <Promise<string>> - the ceremony bucket name.
+ */
+const handleCeremonyBucketCreation = async (firebaseFunctions: Functions, ceremonyPrefix: string): Promise<string> => {
+    // Compose bucket name using the ceremony prefix.
+    const bucketName = getBucketName(ceremonyPrefix, process.env.CONFIG_CEREMONY_BUCKET_POSTFIX!)
+
+    const spinner = customSpinner(`Getting ready for ceremony files and data storage...`, `clock`)
+    spinner.start()
+
+    // Make the call to create the bucket.
+    const created = await createS3Bucket(firebaseFunctions, bucketName)
+
+    if (!created) showError(CORE_SERVICES_ERRORS.AWS_CEREMONY_BUCKET_CREATION, true)
+
+    spinner.succeed(`Ceremony bucket has been successfully created`)
+
+    return bucketName
+}
+
+/**
+ * Upload a circuit artifact (R1CS, zKey, PoT) to ceremony storage bucket.
+ * @dev this method leverages the AWS S3 multi-part upload under the hood.
+ * @param firebaseFunctions <Functions> - the Firebase Cloud Functions instance connected to the current application.
+ * @param bucketName <string> - the ceremony bucket name.
+ * @param storageFilePath <string> - the storage (bucket) path where the file should be uploaded.
+ * @param localPathAndFileName <string> - the local file path where is located.
+ * @param completeFilename <string> - the complete filename.
+ */
+const handleCircuitArtifactUploadToStorage = async (
+    firebaseFunctions: Functions,
+    bucketName: string,
+    storageFilePath: string,
+    localPathAndFileName: string,
+    completeFilename: string
+) => {
+    const spinner = customSpinner(`Uploading ${theme.text.bold(completeFilename)} file to ceremony storage...`, `clock`)
+    spinner.start()
+
+    await multiPartUpload(
+        firebaseFunctions,
+        bucketName,
+        storageFilePath,
+        localPathAndFileName,
+        String(process.env.CONFIG_STREAM_CHUNK_SIZE_IN_MB),
```

Comment:
We never check that these were set in the .env

Reply:
We are testing it in the `bootstrapCommandExecutionAndServices` method here

https://github.com/quadratic-funding/mpc-phase2-suite/pull/320/files/7eddb5d7f2fbd8bde2e13df7b835eeb2cbafb0b4#diff-87f6660d36e6e6644127c9acbc50a32dd9601993eb65311996ffdd1f2e066ed7L40